polar-core/src/partial/simplify.rs

```diff
@@ -149,6 +87,84 @@ impl Folder for PartialToExpression {
     }
 }
 
+impl Simplifier {
+    /// Remove duplicate constraints from a partial.
+    fn deduplicate_constraints(&mut self, partial: Partial) -> Partial {
+        let mut seen: HashSet<&Operation> = HashSet::new();
+        let constraints = partial
+            .constraints()
+            .iter()
+            .filter(|o| seen.insert(o))
+            .cloned()
+            .collect();
+        partial.clone_with_constraints(constraints)
+    }
+
+    /// If ``operation`` is an AND or OR operation with 1 argument, return the 1st argument.
+    ///
+    /// Returns: Some(op) if a rewrite occured, or None.
+    fn maybe_unwrap_single_argument_and_or(&self, operation: &Operation) -> Option<Operation> {
+        match operation {
+            // Unwrap a single-arg And or Or expression and fold the inner term.
+            Operation {
+                operator: Operator::And,
+                args,
+            }
+            | Operation {
+                operator: Operator::Or,
+                args,
+            } if args.len() == 1 => {
+                if let Value::Expression(op) = args[0].value() {
```

Comment:
What could this be if not a `Value::Expression`? Is this just for getting at the inner operation? (I recognize that I'm asking you to diagnose code that I originally wrote 😆)

Reply:
Yeah... I had the same thought.  I think it's invalid Polar if this isn't an expression.  Seems okay like this though.